src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/ModuleInitializerAttribute.cs

```diff
@@ -0,0 +1,36 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Runtime.CompilerServices
+{
+    /// <summary>
+    /// Used to indicate to the compiler that a method should be called
+    /// in its containing module's initializer.
+    /// </summary>
+    /// <remarks>
+    /// When one or more valid methods
+    /// with this attribute are found in a compilation, the compiler will
+    /// emit a `&lt;Module&gt;..cctor` which calls the attributed methods
+    /// in the order they appear in source.
```

Comment:
Are we guaranteeing order of execution?

Reply:
The ordering of module initializers is similar to how the compiler deals with partial declarations. The language specification says that the ordering is undefined but the compiler will always emit them in a deterministic order based on their position in the source files.

Reply:
> The language specification says that the ordering is undefined

Ok, we shouldn't include an implementation detail of the compiler in the documentation for the attribute.  That effectively elevates it to a guarantee of the language.